pallene/driver.lua

```diff
@@ -29,27 +30,24 @@ local function check_source_filename(argv0, file_name, expected_ext)
     return name
 end
 
---
--- Run AST and IR passes, up-to and including the specified pass. This is meant for unit tests.
---
-function driver.compile_internal(filename, stop_after)
-    stop_after = stop_after or "optimize"
-    local err, errs
+function driver.load_input(filename)
+    local input, err = false, nil
```

Comment:
We might want to use the early-exit pattern here to be more consistent with the other functions. In this particular case we can write it in a way that doesn't need a forward declaration for input and err:

```
local base_name, err = check_source(...)
if not base_name then
    return base_name, err
end
return util.get_file_contents(...)
```